tests/specs/mcd/word-pack.k

```diff
@@ -108,4 +177,60 @@ module WORD-PACK-COMMON
 
     rule (UINT48_2 *Int pow208) |Int (maskWordPackAddrUInt48UInt48_3 &Int ADDR_UINT48_UINT48) => #WordPackAddrUInt48UInt48(maxUInt160 &Int ADDR_UINT48_UINT48, maxUInt48 &Int (ADDR_UINT48_UINT48 /Int pow160), UINT48_2) requires #rangeUInt(256, ADDR_UINT48_UINT48) andBool #rangeUInt(48, UINT48_2) [simplification]
     rule (UINT48_1 *Int pow160) |Int (maskWordPackAddrUInt48UInt48_2 &Int ADDR_UINT48_UINT48) => #WordPackAddrUInt48UInt48(maxUInt160 &Int ADDR_UINT48_UINT48, UINT48_1, maxUInt48 &Int (ADDR_UINT48_UINT48 /Int pow208)) requires #rangeUInt(256, ADDR_UINT48_UINT48) andBool #rangeUInt(48, UINT48_1) [simplification]
+
+    rule maxUInt48 &Int (maskWordPackUInt48UInt48_2 &Int A) => 0 requires 0 <=Int A [simplification]
+
+    //
+    // #WordPack components
+    //
+    rule maxUInt160 &Int #WordPackAddrUInt48UInt48(X:Int, _, _) => X [simplification]
+    rule maxUInt48 &Int (#WordPackAddrUInt48UInt48(_, Y:Int, _) /Int pow160) => Y [simplification]
+    rule #WordPackAddrUInt48UInt48(_, _, Z:Int) /Int pow208 => Z [simplification]
+
+    rule maxUInt48 &Int #WordPackUInt48UInt48 (X:Int , _) => X [simplification]
+    rule #WordPackUInt48UInt48 (_, Y:Int) /Int pow48 => Y [simplification]
+
+    rule maxUInt160 &Int #WordPackAddrUInt8 (X:Int , _) => X [simplification]
+    rule #WordPackAddrUInt8 (_, Y:Int) /Int pow160 => Y [simplification]
+
+    //
+    // #WordPack ranges
+    //
+    rule 0 <=Int #WordPackAddrUInt48UInt48(_, _, _)     => true [simplification]
+    rule #WordPackAddrUInt48UInt48(_, _, _) <Int pow256 => true [simplification]
+
+    rule 0 <=Int #WordPackUInt48UInt48(_, _)    => true [simplification]
+    rule #WordPackUInt48UInt48(_, _) <Int pow96 => true [simplification]
+
+    rule 0 <=Int #WordPackAddrUInt8(_, _)     => true [simplification]
+    rule #WordPackAddrUInt8(_, _) <Int pow168 => true [simplification]
+
+    //
+    // More simplifications surrounding pow48 and maxUInt48
+    //
+    rule maxUInt48 &Int X:Int => X requires #rangeUInt(48, X) [simplification]
+    rule maxUInt48 &Int (X /Int pow48) => X /Int pow48 requires #rangeUInt(96, X) [simplification]
+    rule (maxUInt48 &Int _) /Int C => 0 requires C >=Int maxUInt48 [simplification]
+    rule X /Int pow48 => 0 requires #rangeUInt(48, X) [simplification]
+
+    rule maxUInt48 &Int (A:Int |Int B:Int) => (maxUInt48 &Int A) |Int (maxUInt48 &Int B)
+      requires 0 <=Int A andBool 0 <=Int B [simplification]
+
+    rule MAX_UINT &Int X <Int Y => true
+      requires 0 <=Int X
+       andBool MAX_UINT <Int Y
+       andBool 0 <=Int MAX_UINT [simplification]
+
+    //
+    // More simplifications surrounding pow160 and maxUInt160
+    //
+    rule maxUInt160 &Int X:Int => X requires #rangeUInt(160, X) [simplification]
+    rule (maxUInt160 &Int _) /Int C => 0 requires C >=Int maxUInt160 [simplification]
```

Comment:
```suggestion
    rule (maxUInt160 &Int _) /Int C => 0 requires maxUInt160 <=Int C [simplification]
```

Reply:
Same here, shouldn't it be `maxUInt160 <Int C`, not `<=Int`?

Can we add a test of this case?

Also, I think both these lemmas can be generalized to another lemma:

```
    rule (X &Int Y) /Int Z => 0
      requires 0 <=Int X andBool 0 <=Int Y andBool 0 <Int Z
       andBool (Z <Int X orBool Z <Int Y)
      [simplification]
```

Reply:
Yes, that makes sense, although I think the `Z <Int X orBool Z <Int Y` condition should be `X <Int Z orBool Y <Int Z`. I wonder if all the maxUInt160/48 masking lemmas can be similarly generalized.

Reply:
This is also fixed.